pilot/pkg/security/authz/builder/builder.go

```diff
@@ -96,6 +99,16 @@ func New(trustDomainBundle trustdomain.Bundle, push *model.PushContext, policies
 	}
 }
 
+func (b Builder) BuildHTTPAmbient(listener string) []*hcm.HttpFilter {
+	b.AmbientListenerName = listener
```

Comment:
Isn't the builder used beyond a single call? Modifying the entire struct may be problematic?

Reply:
I haven't found it to be problematic.